d3a_interface/matching_algorithms/abstract_matching_algorithm.py

```diff
@@ -0,0 +1,16 @@
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+
+class AbstractMatchingAlgorithm(ABC):
```

Comment:
Minor and not sure, but I would not put "abstract" into the name.
How about `BaseMatchingAlgorithm`